### PR TITLE
Add includesSelf option to Aura rule elements

### PR DESCRIPTION
**Pull request by simonward**

### CODE REVIEW

**Comment by stwlam** [comment]
`includesSelf` defaulting to `true` in case of ally-affecting effects

**Comment by TikaelSol** [comment]
`npm run lint:fix` will fix the linting errors for you.

**Comment by simonward** [comment]
> `npm run lint:fix` will fix the linting errors for you.

Thank you!

File: src/module/actor/base.ts [hunk 1]
```diff
@@ -359,7 +359,11 @@ class ActorPF2e extends Actor<TokenDocumentPF2e, ItemTypeMap> {
                 continue;
             }
 
-            if (data.affects === "allies" && this.isAllyOf(origin)) {
+            if (
+                data.affects === "allies" &&
+                ((data.includesSelf ?? true) || this !== origin) &&
```
Comment (stwlam) [inline src/module/actor/base.ts:364]: This should be guaranteed set by the time it reaches this point.

**Comment by simonward** [comment]
It looks like TokenAura has an unused includesSelf. Should this be removed?

File: src/module/rules/rule-element/aura.ts [hunk 2]
```diff
@@ -94,6 +94,7 @@ export class AuraRuleElement extends RuleElementPF2e {
         effect.affects ??= "all";
         effect.removeOnExit ??= Array.isArray(effect.events) ? effect.events.includes("enter") : false;
         effect.save ??= null;
+        effect.includesSelf ??= true;
```
Comment (stwlam) [inline src/module/rules/rule-element/aura.ts:97]:
```suggestion
        effect.includesSelf ??= effect.affects !== "enemies";
```

**Pull request merged**